docs/JSONDocumentation.md

```diff
@@ -0,0 +1,379 @@
+
+
+#### Switch Section
+
+##### Optional Keys:
+
+| Key        | Type                             | .NET                                    | Java                                         |
+|------------|----------------------------------|-----------------------------------------|----------------------------------------------|
+| Labels     | array of string                  | Case labels, empty indicates 'default'. | Case labels, empty indicates 'default' case. |
+| Statements | array of [Statement](#statement) |                                         |                                              |
+
+
+### Invocation
+
+'$type' key should be equal to: "LivingDocumentation.InvocationDescription, LivingDocumentation.Descriptions".
+
+#### Required keys:
+
+| Key   | Type   | .NET                  | Java                  |
+|-------|--------|-----------------------|-----------------------|
+| $type | const  | Type of the statement | Type of the statement |
+
+#### Optional keys:
+
+| Key             | Type                                              | .NET                                     | Java                                     |
+|-----------------|---------------------------------------------------|------------------------------------------|------------------------------------------|
+| ContainingType  | string                                            | Class that the invoked method belongs to | Class that the invoked method belongs to |
+| Arguments       | array of [AttributeArgument](#attribute-argument) |                                          |                                          |
+
+
+### Assignment
+
+'$type' key should be equal to: "LivingDocumentation.AssignmentDescription, LivingDocumentation.Descriptions".
+
+#### Required keys:
+
+| Key      | Type   | .NET                              | Java                              |
+|----------|--------|-----------------------------------|-----------------------------------|
+| $type    | const  | Type of the statement             | Type of the statement             |
+| Left     | string | Left-hand side of the assignment  | Left-hand side of the assignment  |
+| Operator | string | Assignment operator               | Assignment operator               |
+| Right    | string | Right-hand side of the assignment | Right-hand side of the assignment |
+
+
+### Return
+
+'$type' key should be equal to: "LivingDocumentation.ReturnDescription, LivingDocumentation.Descriptions".
+
+#### Required keys:
+
+| Key        | Type                                       | .NET                  | Java                  |
+|------------|--------------------------------------------|-----------------------|-----------------------|
+| $type      | const                                      | Type of the statement | Type of the statement |
+| Expression | string                                     | Return expression     | Return expression     |
```

Comment:
Is `Expression` indeed required? I would expect this `return` not to have an `Expression`:

```
void foo() {
  return;
}
```

Reply:
This is an interesting one actually. It is required in .NET (sometimes containing the value null). However, this is not the case for Java (as we added a @JsonInclude(Include.NON_NULL) to it).

Reply:
Interesting indeed! Probably good to make this do the same thing on both sides, so:

* Add a note here that the type of this thing is either string or `null`, for `void` methods
* Change the schema to match, if required
* Change our analyzer to match, if required

Reply:
Submitted #111 for  this.